config/env_sample.json

```diff
@@ -239,25 +239,25 @@
         "canvas": {
             "query": [
                 "SELECT 'canvas' AS resource_type, ",
-                "CAST(SUBSTR(JSON_EXTRACT_SCALAR(event, '$.object.id'), 35) AS STRING) AS resource_id, ",
-                "CAST(SUBSTR(JSON_EXTRACT_SCALAR(event, '$.membership.member.id'), 29) AS INT64) AS user_id, ",
-                "cast(null as string) AS user_login_name,",
-                "CAST(SUBSTR(JSON_EXTRACT_SCALAR(event, '$.group.id'), 31) AS INT64) AS course_id, ",
+                "CAST(SUBSTR(object.id, -17) AS STRING) AS resource_id, ",
+                "CAST(SUBSTR(membership.id, -17) AS INT64) AS user_id, ",
+                "cast(null as string) AS user_login_name, ",
+                "CAST(SUBSTR(`group`.id, -17) AS INT64) course_id, ",
                 "COALESCE(",
-                "JSON_EXTRACT_SCALAR(event, '$.object.extensions[\\'com.instructure.canvas\\'].asset_name'), ",
-                "JSON_EXTRACT_SCALAR(event, '$.object.extensions[\\'com.instructure.canvas\\'].filename'), ",
-                "JSON_EXTRACT_SCALAR(event, '$.object.name'), ",
-                "'attachment'",
+                "JSON_EXTRACT_SCALAR(object.extensions, '$[\\'com.instructure.canvas\\'][asset_name]'), ",
+                "JSON_EXTRACT_SCALAR(object.extensions, '$[\\'com.instructure.canvas\\'][filename]'), ",
+                "object.name, ",
+                "'attachment' ",
                 ") as name, ",
                 "datetime(EVENT_TIME) as access_time ",
-                "FROM event_store.events ",
-                "where JSON_EXTRACT_SCALAR(event, '$.edApp.id') IN ",
-                "UNNEST(['http://m.canvas.umich.edu/' , 'http://umich.instructure.com/' ]) ",
+                "FROM event_store.expanded ",
+                "where ",
+                "ed_app.id IN UNNEST(['http://m.canvas.umich.edu/' , 'http://umich.instructure.com/' ]) ",
                 "and type = 'NavigationEvent' ",
-                "and SUBSTR(JSON_EXTRACT_SCALAR(event, '$.object.id'), 24, 10) = 'attachment' ",
-                "and JSON_EXTRACT_SCALAR(event, '$.action') = 'NavigatedTo' ",
-                "and JSON_EXTRACT_SCALAR(event, '$.membership.member.id') is not null ",
-                "and SUBSTR(JSON_EXTRACT_SCALAR(event, '$.group.id'), 31) IN UNNEST(@course_ids) "
+                "and SUBSTR(object.id, 24, 10) = 'attachment' ",
```

Comment:
It looks like this value is the same as `object.name` now, so don't have to do this `SUBSTR` here, can do a straight and `object.name = 'attachment'`

Reply:
updated with object.name